visearch-android/src/main/java/com/visenze/visearch/android/ProductBaseSearchParams.java

```diff
@@ -0,0 +1,470 @@
+package com.visenze.visearch.android;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+import com.visenze.visearch.android.network.RetrofitQueryMap;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ProductBaseSearchParams {
+
+    @SerializedName("page")
+    private Integer page;
+
+    @SerializedName("limit")
+    private Integer limit;
+
+    @SerializedName("va_uid")
+    private String vaUid;
+
+    @SerializedName("va_sid")
+    private String vaSid;
+
+    @SerializedName("facet_limit")
+    private Integer facetLimit;
+
+    @SerializedName("facets_show_count")
+    private Boolean facetsShowCount;
+
+    @SerializedName("sort_by")
+    private String sortBy;
+
+    @SerializedName("group_by")
+    private String groupBy;
+
+    @SerializedName("group_limit")
+    private Integer groupLimit;
+
+    @SerializedName("sort_group_strategy")
+    private String sortGroupStrategy;
+
+    @SerializedName("score")
+    private Boolean score;
+
+    @SerializedName("score_min")
+    private Float score_min;
+
+    @SerializedName("score_max")
+    private Float scoreMax;
+
+    @SerializedName("return_fields_mapping")
+    private Boolean returnFieldsMapping;
+
+    @SerializedName("return_image_s3_url")
+    private Boolean returnImageS3Url;
+
+    @SerializedName("debug")
+    private Boolean debug;
+
+    @SerializedName("vtt_source")
+    private String vttSource;
+
+    @SerializedName("va_sdk")
+    private String vaSdk;
+
+    @SerializedName("va_sdk_version")
+    private String vaSdkVersion;
+
+    @SerializedName("va_os")
+    private String vaOs;
+
+    @SerializedName("va_osv")
+    private String vaOsv;
+
+    @SerializedName("va_device_brand")
+    private String vaDeviceBrand;
+
+    @SerializedName("va_device_model")
+    private String vaDeviceModel;
+
+    @SerializedName("va_app_bundle_id")
+    private String vaAppBundleId;
+
+    @SerializedName("va_app_name")
+    private String vaAppName;
+
+    @SerializedName("va_app_version")
+    private String vaAppVersion;
+
+    @SerializedName("va_aaid")
+    private String vaAaid;
+
+    @SerializedName("va_didmd5")
+    private String vaDidmd5;
+
+    @SerializedName("va_n1")
+    private String vaN1;
```

Comment:
this is a numeric field, change data type to BigDecimal or Double

Reply:
changed already